src/main/java/org/jabref/logic/importer/fetcher/DoiFetcher.java

```diff
@@ -81,7 +81,12 @@ public Optional<BibEntry> performSearchById(String identifier) throws FetcherExc
                     bibtexString = download.asString();
                 } catch (IOException e) {
                     // an IOException will be thrown if download is unable to download from the doiURL
-                    throw new FetcherException(Localization.lang("No DOI data exists"), e);
+                    // dispatch the IOException
+                    if (e instanceof FileNotFoundException) {
+                        throw new FetcherException(Localization.lang("Client Error"), e);
+                    } else {
+                        throw new FetcherException(Localization.lang("Server Error"), e);
```

Comment:
Does it make sense to give those exceptions more meaningful types? Client and Server error seem a bit too arbitrary to me. Also, are exceptions usually localized?

Reply:
I would introduce new exceptions `FetcherClientException extends FetcherException` and `FetcherServerException extends FetcherException`

Reply:
That would be great! If it is finished, could you inform me?